flink-core/src/main/java/org/apache/flink/api/common/state/AppendingState.java

```diff
@@ -46,8 +46,14 @@
 	 * operator instance. If state partitioning is applied, the value returned
 	 * depends on the current operator input, as the operator maintains an
 	 * independent state for each partition.
-	 * 
-	 * @return The operator state value corresponding to the current input.
+	 *
+	 * <p>
+	 *     <b>NOTE TO IMPLEMENTERS:</b> if the state is empty, then this method
+	 *     should return {@code null}.
+	 * </p>
+	 *
+	 * @return The operator state value corresponding to the current input or {@code null}
+	 * is the state is empty.
```

Comment:
Good idea to include this message!

There's a type, though, should be "if the state is empty" 😉